addon/lib/survey.js

```diff
@@ -81,61 +75,34 @@ function checkInstallDate(installDate, addonId) {
   return false;
 }
 
-function launchSurvey(options) {
-  const interval = checkInstallDate(options.installDate, options.addonId);
-  if (typeof interval !== 'number') return;
-
-  const WM = Cc['@mozilla.org/appshell/window-mediator;1'].
-                        getService(Ci.nsIWindowMediator);
-  const win = WM.getMostRecentWindow('navigator:browser');
-  const browser = win.gBrowser;
-  const notifyBox = browser.getNotificationBox();
-
-  const addonId     = options.addonId || 'default_id';
-  const duration    = options.duration || 60; // default is 1 minute.
-  const surveyUrl   = options.surveyUrl || '';
-  const persistence = options.persistence || 0;
-
-  const box = notifyBox.appendNotification(
-    options.label    || 'Default label',
-    options.value    || 'Default Value',
-    options.image    || 'invalid.png',
-    options.priority || notifyBox.PRIORITY_INFO_LOW,
-    options.buttons  || []);
-
-  const messageText = getAnonEl(win, box, 'messageText');
-  const messageImage = getAnonEl(win, box, 'messageImage');
-
-  const ratingFragment = getRatingUI(win, notifyBox, box, messageText,
-                                     addonId, interval, surveyUrl);
-
-  // Make sure the stars are not pushed to the right by the spacer.
-  const rightSpacer = win.document.createElement('spacer');
-  rightSpacer.flex = 20;
-  ratingFragment.appendChild(rightSpacer);
-
-  messageText.flex = 0; // Collapse the space before the stars.
-  const leftSpacer = messageText.nextSibling;
-  leftSpacer.flex = 0;
-
-  box.appendChild(ratingFragment);
-
-  box.classList.add('heartbeat');
-  box.persistence = persistence;
-  messageImage.classList.add('heartbeat', 'pulse-onshow');
-  messageText.classList.add('heartbeat');
-  messageImage.setAttribute('style', 'filter: invert(80%)');
-
-  // End the survey if the user hasn't responded before expiration.
-  if (!options.privateWindowsOnly) {
-    setTimeout(() => {
-      notifyBox.removeNotification(box);
-    }, 1000 * duration);
-  }
+function launchSurvey(experiment, interval) {
+  showRating({ experiment })
+    .then(
+      rating => {
+        const urlParams = {
+          id: experiment.addon_id,
+          installed: Object.keys(store.installedAddons),
+          surveyClicked: false,
+          rating,
+          interval
+        };
+        return showSurveyButton({ experiment })
```

Comment:
Yeah, I generally agree, and I thought about doing it that way too. The trivial reason I went this way was so that the `rating` from `showRating` wouldn't need to be null checked. Then to be consistent with that I went the same way with `showSurveyButton`... A *VERY* trivial difference.